Fixes #782: Add `Yara_Scan_ATM_MALWARE` conf and rules

CypherpunkSamurai opened the pull request:
Added new Yara rules from fboldewin/YARA-rules to solve #782

# Description

This PR adds more Yara rules from [fboldewin/YARA-rules](https://github.com/fboldewin/YARA-rules) as mentioned in issue 782.

## Related issues
new Yara rule repo to add #782

## Type of change
- [X] New feature (non-breaking change which adds functionality).

# Checklist

- [x] The pull request is for the branch develop
- [x] A new analyzer or connector was added, in which case:
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [x] [Advanced-Usage](./Advanced-Usage.md) was updated (in case the analyzer/connector provides additional optional configuration).
    - [x] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [x] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in it's class to apply the necessary decorators.
    - [x] If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [x] If external libraries/packages with restrictive licenses were used, they were added in the [Legal Notice](https://github.com/certego/IntelOwl/blob/master/.github/legal_notice.md) section.
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [x] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)
  
### Important Rules
- If your changes decrease the overall tests coverage (you will know after the Codecov CI job is done), you should add the required tests to fix the problem
- Everytime you make changes to the PR and you think the work is done, you should explicitly ask for a review

# Real World Example

Please delete if the PR is for bug fixing.
Otherwise, please provide the resulting raw JSON of a finished analysis (and, if you like, a screenshot of the results). This is to allow the maintainers to understand how the analyzer works.

mlodic commented:
Hey,

please you need to read the documentation and the rules about [how to contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) before submitting your PR.

Plus, the PR checklist is not there randomly: it is there to be read and compiled and you did not.

You make the contributors waste their time if you do not follow the rules.

Thank you

eshaan7 commented on configuration/analyzer_config.json:
```diff
@@ -3687,6 +3687,40 @@
       }
     }
   },
+  "ATM Malware": {
```
rename to `Yara_Scan_ATM_MALWARE` and place it above the `Yara_Scan_Community` one.

CypherpunkSamurai replied:
Done. I've made these changes sir. Please review :)

CypherpunkSamurai commented:
> Hey,
> 
> please you need to read the documentation and the rules about [how to contribute](https://intelowl.readthedocs.io/en/latest/Contribute.html) before submitting your PR.
> 
> Plus, the PR checklist is not there randomly: it is there to be read and compiled and you did not.
> 
> You make the contributors waste their time if you do not follow the rules.
> 
> Thank you

Ok. I'm really sorry, I'm new to contributions and just getting started. Please excuse my mistakes.

I'll fix the checklist and add suggested changes.

CypherpunkSamurai commented:
Hello Sir, I've made the requested changes to the previous commit. I've also checked the PR checklist and added respective usage info and test entry. Please check if everything is ok :)

(as a sidenote should I squash the changes into one?)

CypherpunkSamurai commented:
Hello Mr. @mlodic and Mr. @eshaan7. Please review if this PR is ok.

I've added:
* added bootstrap script lines to clone atm malware yara rules
* added `usage.md` entry.
* added yara entry to test_file_scripts.py. (No tests configured yet, however).

mlodic commented:
there is a problem in the tests regarding the urlhaus analyzer. I'll fix it then you'll need to pull the changes from develop to have the tests working correctly

CypherpunkSamurai commented:
> there is a problem in the tests regarding the urlhaus analyzer. I'll fix it then you'll need to pull the changes from develop to have the tests working correctly

Ok Sir, until then I'll try working on the other issues.

mlodic commented:
please pull from develop so the tests can appropriately restart with the right code

mlodic commented:
as you can see in the build, the test says that your configuration does not work. Please try it out locally. See the doc here: https://intelowl.readthedocs.io/en/latest/Tests.html#run-tests-for-a-particular-analyzer-or-class-of-analyzers

CypherpunkSamurai commented:
> as you can see in the build, the test says that your configuration does not work. Please try it out locally. See the doc here: https://intelowl.readthedocs.io/en/latest/Tests.html#run-tests-for-a-particular-analyzer-or-class-of-analyzers

Ok Sir.

There's one problem though, this commit only issued fixes to running IntelOwl correctly with the new rules, and no test cases were added (because i was finding malware samples for all those Yara files). I see you have merged it. Is it required that we add samples?

mlodic commented:
There is no need to find the samples that match the signatures in that repo.

Based on how IntelOwl is built, it was enough to add the new analyzer name in the `tests/analyzers_manager/test_file_scripts.py` file and IntelOwl would use these rules.  With or without matches we do not care, we just care that we correctly use those rules.

Thank you
